### PR TITLE
Miniapp: token gated casts

### PR DESCRIPTION
**Pull request by davidfurlong**
**There's a blocker right now for releasing this Miniapp:**
Lit protocol is upgrading to v3 (mainnet) which is called Habanero, and is scheduled for December. The v3 testnet is not persistent, is unstable, and has breaking changes every couple days, requiring a node package update to fix.
   
Other blockers: 
- hot reloading breaks the encryption endpoint locally (somewhere in the lit library), tbd what happens in prod.
- unclear whether signature needs to have the same chainid as the access control rules (maybe not)
- need to choose a url for the production gateway

Somewhat relies on the new gateway: https://github.com/mod-protocol/gateway

### CODE REVIEW

**Review by stephancill — CHANGES_REQUESTED** [review]
Looking good. Left some notes

File: docs/pages/metadata-cache.mdx [hunk 1]
```diff
@@ -74,7 +76,74 @@ Returned metadata objects conform to the `UrlMetadata` type. This can then be us
...
-  return <RenderEmbed embed={embedData} />
-})
-```
\ No newline at end of file
+  const embedData: UrlMetadata = metadataResponse[cast.hash][index];
+  return <RenderEmbed embed={embedData} />;
+});
+```
+
+## `/api/cast-embeds-metadata/by-url`
```
[the remark this answers is not in the document]
Comment (davidfurlong) [reply docs/pages/metadata-cache.mdx:84]: agreed, but that means no backwards compat; will have to let integrators know

[the remark this answers is not in the document]
Comment (davidfurlong) [reply docs/pages/metadata-cache.mdx:84]: agreed, but that means no backwards compat; will have to let integrators know

File: examples/api/src/app/api/lit-protocol/route.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+      contract,
+      standardContractType,
+      tokens,
+    });
+  const litNodeClient = await getLitNodeClient();
+
+  // 1. Encryption
+  // <Blob> encryptedString
+  // <Uint8Array(32)> dataToEncryptHash`
+  const { ciphertext, dataToEncryptHash } = await LitJsSdk.encryptString(
```
[the remark this answers is not in the document]
Comment (davidfurlong) [reply examples/api/src/app/api/lit-protocol/route.ts:78]: yeah, also Mini-apps can't load arbitrary JS today

[the remark this answers is not in the document]
Comment (davidfurlong) [reply examples/api/src/app/api/lit-protocol/route.ts:78]: theoretically this backend function here could be run a secure JS env on a client; not sure how much their lib depends on node apis directly

File: packages/core/src/manifest.ts [hunk 3]
```diff
@@ -116,6 +122,33 @@ type OpenLinkAction = BaseAction & {
   url: string;
 };
 
+export type EthPersonalSignData = {
+  // domain: string;
+  // address: string;
+  statement: string;
+  // uri: string;
```
[the remark this answers is not in the document]
Comment (davidfurlong) [reply packages/core/src/manifest.ts:129]: the client provides these in their `onEthPersonalSign` resolver; will remove

File: miniapps/lit-protocol/src/action.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+const action: ModElement[] = [
+  {
+    type: "vertical-layout",
+    elements: [
+      {
+        type: "textarea",
+        ref: "plaintext",
+        placeholder: "Content to token gate",
+      },
```
[the remark this answers is not in the document]
Comment (davidfurlong) [reply miniapps/lit-protocol/src/action.ts:11]: good question - the cast contents includes mentions (that won't work) and special renderers (like textcuts) that Mini-apps can't directly use. Additionally casts include embeds that we can't easily render if we added them to the encrypted contents, and token gated casts would probably still include or be able to include the cast body; so it'd be a bit confusing to have the cast text appearing/disappearing, or for it to somehow be encrypted when submitted.

In short, I'm not sure our abstractions are particularly close to supporting it in a graceful manner
